### DISCUSSION TITLE
How to track future bills before they arrive?

### DISCUSSION THREAD

**Discussion by PiotrEsse** [firefly-iii/firefly-iii#7155] [Firefly III]
How to show that someone should pay but hasn't paid yet? 

That is, for example, if I rent an apartment and someone is supposed to pay me the rent on the 1st day of each month. Is there a way to have a bill issued to this person on the first of each month but marked as unpaid, and I would only manually or using automatic rules mark it as paid at the time of receiving the transfer. 

Currently, bills work in such a way that when the bill is issued, the transfer to the account is immediately credited, which is not true. 

I understand that Firefly is not a program for invoices or settlements with others, but it is so advanced that I just want to know if it is possible ;-D

**Comment by JC5** [comment]
This is literally what bills are for, what are you doing in regards to:

> Currently, bills work in such a way that when the bill is issued, the transfer to the account is immediately credited, which is not true.

This is not how it works, what is your current way of working?

**Comment by PiotrEsse** [comment]
Ach, James, sorry, I need to check this again as You are right - Ive messed something ;-)

**Comment by PiotrEsse** [comment]
Dear James, 
I am testing matching bills with transaction but I am clueless why it wont work. 

I want to issue monthly a bill for 650zł and match it with a transaction for that same amount. 

This is what I am doing: 
1. Ive created a Bill2. 
![obraz](https://user-images.githubusercontent.com/42764623/233068584-7ae42af5-8648-4233-82f9-ede84db46e78.png)
2. Ive created a automatic rule
![obraz](https://user-images.githubusercontent.com/42764623/233069065-fefb5fa6-be72-4ad7-87c6-f351deb6861c.png)
3. When I seach for transactions FireFly correctly show me 2 transactions are in line with my triggers:
![obraz](https://user-images.githubusercontent.com/42764623/233069495-333f1f65-6eb3-47b1-81b4-13990598f0d8.png)
4. When importer runs and import transactions - nothing happened - no transaction is linked to a bill
5.  When I manually run a rule I am getting following messsage. Seems like it didnt found any transaction.
![obraz](https://user-images.githubusercontent.com/42764623/233069871-b7ead544-c25d-41c8-a70e-31f1e9eef6ab.png)

Do I need to play with dates?
Ive changed trigger method from transaction created to transaction update and it didnt helped. 
 Could You help?

**Reply by JC5** [reply]
Were the transactions already linked to the bill perhaps?

**Reply by PiotrEsse** [reply]
Hi, No, bolls dont have any transaction linked.
![obraz](https://user-images.githubusercontent.com/42764623/233632575-6faa6b45-e633-4123-b755-cc9fac641a33.png)

BUT, Ive read again documentation and found that Bills can only be linked to "withdrawal" and I am trying to mach deposit.

If this goes that way - how can I check if somebody paid me ie: rent for my flat (I am renting a flat)

**Reply by JC5** [reply]
> If this goes that way - how can I check if somebody paid me ie: rent for my flat (I am renting a flat)

You'll have to monitor for the transaction itself, there is no such thing as bills for income I'm afraid.

**Reply by PiotrEsse** [reply]
Thank You, make sense.
